src/server/http/__tests__/xsrf.js

```diff
@@ -104,14 +144,58 @@ describe('xsrf request filter', function () {
         expect(resp.payload).to.be('ok');
       });
 
-      it('rejects requests without either an xsrf or version header', async function () {
+      it('accepts requests with any allowed media type', async function () {
+        const allowedContentTypes = [
```

Comment:
I think your hesitation is warranted.  We need to be in as much control of the behavior here as possible since it's such a critical security feature, so that hapi thinks it is OK to ignore the content-type under any circumstance is enough of a reason to not use that particular hapi feature.